implementations/micrometer-registry-otlp/src/test/java/io/micrometer/registry/otlp/OtlpConfigTest.java

```diff
@@ -100,8 +101,17 @@ void headersConfigTakesPrecedenceOverEnvVars() throws Exception {
     @Test
     void headersUseEnvVarWhenConfigNotSet() throws Exception {
         OtlpConfig config = k -> null;
-        withEnvironmentVariable("OTEL_EXPORTER_OTLP_HEADERS", "header2=value")
-            .execute(() -> assertThat(config.headers()).containsEntry("header2", "value").hasSize(1));
+        withEnvironmentVariable("OTEL_EXPORTER_OTLP_HEADERS", "header2=va%20lue")
+            .execute(() -> assertThat(config.headers()).containsEntry("header2", "va lue").hasSize(1));
```

Comment:
Let's also add a test to verify that unencoded values continue to work since they did before.
`header2=va lue` should continue to work.

Reply:
@shakuzen done